### DISCUSSION TITLE
Is it possible to only show the menu while a key is held?

### DISCUSSION THREAD

**Discussion by CarlQLange** [Schneegans/Fly-Pie#218] [Fly-Pie Q&A]
I've got a menu set up to display when I press some arbitrary number of keys, and a mouse button macro'd to press those keys and bring up the menu. I'd love for the menu to dismiss when I stop holding the button (eg, I have changed my mind about doing anything), is that possible? Alternately, if the menu could dismiss if I press the shortcut again, that would work too.

**Comment by Schneegans** [comment]
In fact, I am not sure :laughing:. Probably it could be done somehow. There is the `ToggleMenu` D-Bus command which you can use to open _and_ close one of your menus. Like this:

```bash
gdbus call --session --dest org.gnome.Shell --object-path /org/gnome/shell/extensions/flypie \
           --method org.gnome.Shell.Extensions.flypie.ToggleMenu 'Main Menu'
```

If you can somehow manage to execute this command on a mouse button press, it could work indeed.

Alternatively, we could consider closing the active menu when its hotkey is pressed again. However I will have to think carefully about this as it may interfere with the Marking Mode and the Turbo Mode...
